torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -85,6 +85,7 @@ void PropagateShapeOnNode(Node * node) {
       int64_t dim = node->i(kdim);
       int64_t length = node->i(klength);
       sizes.at(dim) = length;
+      // xxx - does not propagate strides of narrow correctly
       node->output()->setType(tp->withSizes(sizes));
```

Comment:
Can't you just replace it with `tp->withSizesAndStrides(sizes, tp->strides())`?

Reply:
Is that correct? I guess narrow does not change the strides, just the start pointer and the size.

Reply:
Yes, strides will remain the same